### PR TITLE
with context

### PR DESCRIPTION
**Pull request by WAbuAwwad**

### CODE REVIEW

File: src/components/home/home.tsx [hunk 1]
```diff
@@ -65,24 +64,27 @@ class Home extends Component<RouteComponentProps> {
     this.setState({ tab: value });
   };
   render() {
-    return (
+    return this.state.isLoading ? (
+      <img src="./805.gif" alt="loading..." />
```
Comment (Abdallatif) [inline src/components/home/home.tsx]: is the loader for the home page only? I think we can use this every where in the app

Comment (WAbuAwwad) [reply src/components/home/home.tsx]: I I think it's not that slow to show it . I can put it

File: src/context.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import * as React from "react";
+
+const context = React.createContext({
```
Comment (Abdallatif) [inline src/context.tsx]: let's call this `userContext`

File: src/components/menu/menu.tsx [hunk 3]
```diff
@@ -29,35 +17,44 @@ class Menu extends Component {
...
-                </Button>
-                <Button color="inherit" onClick={() => navigate("sign-in")}>
-                  Sign in
-                </Button>
-                <Button color="inherit" onClick={() => navigate("sign-up")}>
-                  Sign up
-                </Button>
-              </div>
-            )}
+            <Consumer>
```
Comment (Abdallatif) [inline src/components/menu/menu.tsx]: this would be more readable `<UserContext.Consumer>`

File: src/App.tsx [hunk 4]
```diff
@@ -7,18 +7,34 @@ import NewArticle from "./components/new article/newArticle";
...
+import Menu from "./components/menu/menu";
 
 class App extends Component<RouteComponentProps> {
+  state = {
+    isLoggedIn: false,
+    username: ""
+  };
+
   render() {
+    checkLoggedIn().then(data => {
```
Comment (Abdallatif) [inline src/App.tsx]: you shouldn't do side effects in render function

File: src/components/home/home.tsx [hunk 5]
```diff
@@ -23,24 +23,24 @@ class Home extends Component<RouteComponentProps> {
...
-        this.setState({ isLoggedIn: true });
-        fetchFeedArticles(this.state.page).then(data => {
-          this.setState({ feedArticles: data });
-        });
-      }
+    fetchArticles(this.state.page)
+      .then(data => {
+        this.setState({ articles: data });
+      })
+      .finally(function() {
```
Comment (Abdallatif) [inline src/components/home/home.tsx]: use arrow function, so you don't need to use `that`

File: src/components/menu/menu.tsx (hunk 3 above)
Comment (WAbuAwwad) [reply src/components/menu/menu.tsx]: we import the consumer directly

Comment (Abdallatif) [reply src/components/menu/menu.tsx]: Can you rename the consumer? so if we have more consumers it would be easier to distinguish between them

**Pull request merged**